src/utils/constants_from_env.rs

```diff
@@ -0,0 +1,344 @@
+use crate::utils::from_env::{EnvItemInfo, FromEnv, FromEnvErr, FromEnvVar};
+use alloy::primitives::{hex::FromHexError, Address};
+use signet_constants::{
+    HostConstants, ParseChainError, PredeployTokens, RollupConstants, SignetConstants,
+    SignetEnvironmentConstants, SignetSystemConstants,
+};
+use std::{borrow::Cow, num::ParseIntError};
+
+/// EnvItemInfo for .env variable holding chain name
+/// Used to implement FromEnv for SignetConstants type structs
+/// that can be instantiated from a single chain name
+const CHAIN_NAME: EnvItemInfo = EnvItemInfo {
+    var: "CHAIN_NAME",
+    description: "The name of the chain, e.g. `pecorino`. If CHAIN_NAME is present, the known, hard-coded constants for the chain will be loaded from the SDK. If CHAIN_NAME is not present, each constant will be loaded from environment variables.",
+    optional: true,
+};
+
+// --- RollupConstants ---
+const ROLLUP_CHAIN_ID: &str = "ROLLUP_CHAIN_ID";
+const ROLLUP_BASE_FEE_RECIPIENT: &str = "ROLLUP_BASE_FEE_RECIPIENT";
+const ROLLUP_ORDERS: &str = "ROLLUP_ORDERS";
+const ROLLUP_PASSAGE: &str = "ROLLUP_PASSAGE";
+const ROLLUP_USDC: &str = "ROLLUP_USDC";
+const ROLLUP_USDT: &str = "ROLLUP_USDT";
+const ROLLUP_WBTC: &str = "ROLLUP_WBTC";
```

Comment:
not a nit for this pr necessarily, but i find we use `RU` and `ROLLUP` interchangeably on bin-base and the SDK, maybe we should consolidate